src/main/java/duke/task/TaskList.java

```diff
@@ -0,0 +1,151 @@
+    }
+
+    /**
+     * Constructs a new TaskList containing a given task list.
+     *
+     * @param taskList the given List of task.
+     */
+    public TaskList(List<Task> taskList) {
+        this.taskList = taskList;
+    }
+
+    /**
+     * Executes the command on the list of tasks.
+     *
+     * @param commands an Array of String which contains command and its arguments.
+     * @param ui       the Ui to deal with the interactions with users.
+     * @param storage  the Storage to deals with loading tasks from the file and saving tasks in the file.
+     * @throws DukeException thrown if the storage process goes wrong.
+     */
+    public void runCommand(String[] commands, Ui ui, Storage storage) throws DukeException {
+        switch (commands[0]) {
+        case "list": {
+            printList(ui);
+            break;
+        }
+        case "find": {
+            find(ui, commands[1]);
+            break;
+        }
+        case "done": {
+            markAsDone(ui, Integer.parseInt(commands[1]) - 1);
+            storage.store(taskList);
+            break;
+        }
+        case "delete": {
+            delete(ui, Integer.parseInt(commands[1]) - 1);
+            storage.store(taskList);
+            break;
+        }
+        case "todo":
+        case "deadline":
+        case "event":
+            addTask(ui, commands[0], commands[1], commands[2]);
+            storage.store(taskList);
+            break;
+        default:
+        }
+    }
+
+    /**
+     * Prints the list of task through the Ui.
+     *
+     * @param ui the Ui to deal with the interactions with users.
+     */
+    public void printList(Ui ui) {
+        ui.printList(this.taskList);
+    }
+
+    /**
+     * Marks done the task in the list of task of a certain index and print through the Ui.
+     *
+     * @param ui    the Ui to deal with the interactions with users.
+     * @param index the index of the task in the task list.
+     */
+    public void markAsDone(Ui ui, int index) {
+        Task task = taskList.get(index);
+        task.markAsDone();
+        ui.printDone(task);
+    }
+
+    /**
+     * Deletes the task in the list of task of a certain index and print through the Ui.
+     *
+     * @param ui    the Ui to deal with the interactions with users.
+     * @param index the index of the task in the task list.
+     */
+    public void delete(Ui ui, int index) {
+        Task task = taskList.get(index);
+        taskList.remove(index);
+        ui.printDelete(task, taskList.size());
+    }
+
+
+    /**
+     * Finds the tasks containing the given keyword in their description and print through the Ui.
+     *
+     * @param ui  the Ui to deal with the interactions with users.
+     * @param key the keyword used to find task in task list.
+     */
+    public void find(Ui ui, String key) {
+        List<Task> result = new ArrayList<>();
+        for (Task task : taskList) {
+            if (task.description.contains(key)) {
```

Comment:
Be careful with using contains here because it might lead to unexpected search results! Take for example, if I did something like this: 
String input = "basement";
input.contains(base) will return true which might not be intended!